### DISCUSSION TITLE
cmake find_path help

### DISCUSSION THREAD

**Discussion by juha-h** [baresip/baresip#2210] [Q&A] [answered]
I got my libbaresip-android baresip build converted from make to cmake except amr module.

 I have two amr source dirs amr and vo-amrwbenc. As a starter, I have tried to make FindAMR.cmake to find NB_INCLUDE_DIR:
```
find_path(NB_INCLUDE_DIR
  NAMES interf_dec.h interf_enc.h
  HINTS "${AMR_INCLUDE_DIR}"
  PATHS /usr/local/include/opencore-amrnb /usr/include/opencore-amrnb
)

message(NB_INCLUDE_DIR ${NB_INCLUDE_DIR})
```
I have this cmake argument:
```
-DAMR_INCLUDE_DIR=$(PWD)/amr/include/opencore-amrnb \
```
and cmake at start tells it value:
```
-DAMR_INCLUDE_DIR=/usr/src/libbaresip-android/amr/include/opencore-amrnb \
```
For sure the two .h files exist in that directory:
```
$ ls /usr/src/libbaresip-android/amr/include/opencore-amrnb
interf_dec.h  interf_enc.h
```
Still cmake does not find the directory, since message prints:
```
NB_INCLUDE_DIRNB_INCLUDE_DIR-NOTFOUND
```
Any hints why the include dir is not found?  I have spend a few hours on this w/o success.

**Comment by sreimers** [comment]
cmake has a debug find option, it should show more details about config and tried paths:

```
cmake --debug-find -B build
```

**Comment by juha-h** [comment]
I'll try debug soon, but I have started to suspect that in this environment cmake is executed in some kind of sandbox and find_path does not find anything outside of it.

**Comment by juha-h** [comment]
Just to add to the previous, NB_INCLUDE_DIR is not found even when I edited FindAMR.cmake like this:
```
find_path(NB_INCLUDE_DIR
  NAMES opencore-amrnb
  PATHS /usr/src/libbaresip-android/amr/include
)
```
and for sure opencore-amrnb exists in the given path.

**Comment by sreimers** [comment, marked as answer]
The android toolchain sets `CMAKE_SYSROOT` which has several effects. To add more search paths you can use/extend `CMAKE_FIND_ROOT_PATH`

**Comment by juha-h** [comment]
Thanks, `CMAKE_FIND_ROOT_PATH=/opt/Android/ndk/25.1.8937393`.  I'll try to add /usr/src/libbaresip-android to it.

**Comment by juha-h** [comment]
I got amr module built when I added this to cmake:
```
-DCMAKE_FIND_ROOT_PATH="$(CMAKE_FIND_ROOT_PATH);$(PWD)/amr;$(PWD)/vo-amrwbenc"
```
Thanks again for help.
